pkg/controllers/leaderworkerset_controller.go

```diff
@@ -109,6 +109,11 @@ func (r *LeaderWorkerSetReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return ctrl.Result{}, err
 	}
 
+	// lws is in deleting and leader sts is also in deleting, return directly.
+	if lws.DeletionTimestamp != nil && (leaderSts == nil || leaderSts.DeletionTimestamp != nil) {
+		return ctrl.Result{}, nil
+	}
+
```

Comment:
Hi @congcongke,

I'm not entirely sure, but could you please clarify why we need to check `(leaderSts == nil || leaderSts.DeletionTimestamp != nil)`? Wouldn't it suffice to just check `lws.DeletionTimestamp != nil` to avoid reconciling LWS that are marked for deletion?

Additionally, do you think it would make more sense to place this piece of code before `leaderSts, err := r.getLeaderStatefulSet(ctx, lws)`?

Reply:
yes, you are right!